src/messaging/utils.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { ICType } from '@kiltprotocol/types'
+import { IMessage, MessageBody } from '../types'
+import * as MessageError from './Error'
+import { CType } from '@kiltprotocol/sdk-js'
+import { UUID } from '@kiltprotocol/utils'
+
+/**
+ * Constructs a message from a message body.
+ * This should be encrypted with [[encrypt]] before sending to the receiver.
+ *
+ * @param body The body of the message.
+ * @param sender The DID of the sender.
+ * @param receiver The DID of the receiver.
+ * @returns The message created.
+ */
+export function fromBody(body: MessageBody, sender: IMessage['sender'], receiver: IMessage['receiver']): IMessage {
```

Comment:
```suggestion
export function fromBody<Body extends IMessageBodyBase>(body: Body, sender: IMessage['sender'], receiver: IMessage['receiver']): IMessage<Body> {
```

Reply:
and I was thinking this could maybe live on the Message.ts file instead? @weichweich what do you think ?

Reply:
toda: [3969f72](https://github.com/KILTprotocol/kilt-extension-api/pull/18/commits/3969f723cc55bd834d102e1261fe33b4b2246d64)